api/units.go

```diff
@@ -76,6 +76,7 @@ func (ur *unitsResource) set(rw http.ResponseWriter, req *http.Request, item str
 	}
 
 	var su schema.Unit
+	newUnit := false
```

Comment:
Better to introduce the variable closer to its use -- in this case, immediately before the conditional block that sets it.